### PR TITLE
Largest number #136

### LINKED ISSUE #136: Largest Number
### Largest Number

Given a list of non-negative integers **_nums_**, arrange them such that they form the **largest number** and return it.
Since the result may be very large, so you need to **return a string** instead of an integer.

**Example 1:**
**Input:** nums = [10,2]
**Output:** "210"
**Explanation:** Order of elements taken in the answer is 2, 10.

**Example 2:**
**Input:** nums = [3,30,34,5,9]
**Output:** "9534330"
**Explanation:** Order of elements taken in the answer is 9, 5, 34, 3, 30.

**Hints:** Array, String, Sorting

### PR DESCRIPTION
**Pull request by ItsPranavz**
#136 Couldn't find a feasible way to convert vector to string without getting 100+ errors but it's getting the job done without needing to print an integer.

### CODE REVIEW

**Comment by Ishwarendra** [comment]
Your logic is incorrect @ItsPranavz. As mentioned by @iamakashrout [here](https://github.com/PEC-CSS/graveyard/issues/136#issuecomment-1454510655), your code doesn't even pass the samples.

**Comment by ken1000minus7** [comment]
@ItsPranavz any updates?

**Comment by ItsPranavz** [comment]
> @ItsPranavz any updates?

I had implemented provision to first break the inputs into digits and then add to vector and also convert the vector to string using to_string and a for loop. However soon after celebrating I realized vector was the wrong data type to store in because you can't push back to the same integer when the last int is same as current int so now I have to rethink everything. 😭

**Comment by ItsPranavz** [comment]
Please review the code now. I shifted back to python after trying my best with cpp but I think there is much to learn still. I tried to use vector but failed and then also tried to store the number of unique digits in an array of length 10 as a frequency distribution at corresponding indices but failed. I guess I need to learn more about data structures in cpp before I try my hand at this.

**Comment by ItsPranavz** [comment]
Python it is for now!

**Comment by Ishwarendra** [comment]
@ItsPranavz incorrect logic. Check the case below
```
1
23
```

**Comment by ken1000minus7** [comment]
@ItsPranavz Your code doesn't pass the 2nd sample, reread the ques you are mixing digits of different numbers

**Comment by ItsPranavz** [comment]
> @ItsPranavz Your code doesn't pass the 2nd sample, reread the ques you are mixing digits of different numbers

Yes, I am understanding my error now I misunderstood the statement the whole time. I am sorry for bothering you again and again. I will fix it for sure this time.

**Comment by ItsPranavz** [comment]
Ok, it has to work this time. I created a score called yield score that tells the relative yield of placing a number of any length calculated as that number divided by the largest number of same length and then sort the samples in decreasing order of this yield score and display in order.

I got this inspiration from the fact that numbers with repeating digits like, 3, 33, 333... have an equal preference to positioning so I thought of using the ratio to the largest n length number as 3/9, 33/99, 333/999 to find a number which has a direct relation to positional order or yield as I call it. So out of 34,3,31 : 34 (34/99) has more yield than 3 (3/9) and then 31 (31/99) has least yield.

I hope my logic is correct now. Fingers crossed.🤞

File: cp/largest-number.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  temp = []
+  order = sorted(map.keys())
+  j = len(order)-1
+  while j >= 0:
+    temp.extend(map[order[j]])
+    j -= 1
+  return "".join(temp)
+
+inp = input().strip("[").strip("]").split(",")
+print(yield_map(inp))
```
Comment (Ishwarendra) [inline cp/largest-number.py]: fix this rest ok
```suggestion
```

**Review by Ishwarendra — CHANGES_REQUESTED** [review]

**Comment by Ishwarendra** [comment]
@ItsPranavz Still incorrect. Check the case below:
```
Input
923 83 8 698 667

Correct Output: (Added space so you can see each number)
923 83 8 698 667 

Your output: 
923883698667
```

**Comment by ItsPranavz** [comment]
Made the change.

**Pull request merged**